detox/src/devices/drivers/SimulatorDriver.js

```diff
@@ -105,6 +107,20 @@ class SimulatorDriver extends IosDriver {
     await this.emitter.emit('shutdownDevice', { deviceId });
   }
 
+  async getViewportSize(udid) {
+    const temporaryFilePath = tempfile('.png');
+
+    await this._applesimutils.takeScreenshot(udid, temporaryFilePath)
```

Comment:
As discussed in the comments, this is a bad way of calculated results. It should come from native.